clustering/common/src/main/java/org/jboss/as/clustering/controller/CredentialSourceDependency.java

```diff
@@ -89,11 +90,16 @@ public CredentialSource get() {
             return this.dependencies.iterator();
         }
 
-        @Deprecated
         @Override
         public ServiceBuilder<Object> addDependency(ServiceName serviceName) {
             this.dependencies.add(new ServiceDependency(serviceName));
             return this;
         }
+
+        @Override
+        public Supplier<Object> requires(ServiceName serviceName) {
+            this.dependencies.add(new ServiceDependency(serviceName));
+            return null;
+        }
```

Comment:
This would cause a NPE if the CredentialSource code were to be modified to utilize this method.

Reply:
CrecendialSource is under our control and it doesn't utilize this method.
I wasn't aware of ServiceSupplierDependency class in clustering.
Your enhanced PR addresses my problem, so I'm closing this pull request.